domain/src/main/java/org/jboss/set/mjolnir/archive/domain/UserRemoval.java

```diff
@@ -25,13 +25,16 @@
 @NamedQueries({
         @NamedQuery(name = UserRemoval.FIND_REMOVALS_TO_PROCESS,
                 query = "SELECT r FROM UserRemoval r WHERE r.started IS NULL" +
-                        " AND (remove_on <= CURRENT_DATE or remove_on IS NULL)")
+                        " AND (remove_on <= CURRENT_DATE or remove_on IS NULL)"),
+        @NamedQuery(name = UserRemoval.FIND_FINISHED_REMOVALS,
+                query = "SELECT r FROM UserRemoval r WHERE (r.completed > :jobStart AND r.status != 'STARTED') OR r.status IN ('FAILED', 'USER_UNKNOWN') ORDER BY r.status DESC, r.username")
```

Comment:
> "(r.completed > :jobStart AND r.status != 'STARTED')"

In theory, when "completed" is set that should imply that status != 'STARTED'.

> "(r.completed > :jobStart AND r.status != 'STARTED') OR r.status IN ('FAILED', 'USER_UNKNOWN')"

Will we not report all FAILED and UNKNOWN removals repeatedly forever this way?

Reply:
Yes, exactly, but I think that in the case of UNKNOWN_USER it is justified and FAILED case should be re-invoked. But I don't have no problem with just reporting current failures.

Reply:
Ah, UNKNOWN_USERs can be safely ignored. We only care about users who registered into Mjolnir and registered their GH username (i.e. those are not UNKNOWN_USER). In reality most user removals will end up to be UNKNOWN_USERs and we don't need to take any notice of those. They may or may not appear in the reports, but should not appear repeatedly. It may be better to omit them to make reports briefer.

Regarding FAILED removals, re-invocation could be implemented in two ways:
* either recycling the same removal record (setting started and state columns to null),
* or inserting new record.

I would prefer the second, because that way we would keep all the information about previous failure, without further complicating the database model.

Reply:
OK, that makes thing more easier. I will change the select accordingly then.